.github/workflows/plugin-compiler-build.yml

```diff
@@ -0,0 +1,105 @@
+---
+name: Plugin-compiler build
+
+on:
+  workflow_run:
+    workflows:
+      - Release
+    types:
+      - completed
+
+env:
+  GOLANG_CROSS: 1.19-bullseye
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ fromJson(steps.verify-tag.outputs.result) && fromJson(steps.verify-tag.outputs.result).tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.workflow_run.head_branch }}
+
+      - name: Set up buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Setup node to match github-script v6 node runtime
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: install dependency
+        run: npm install semver
+
+      - name: Verify whether valid semver tag
+        id: verify-tag
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const semver = require('semver')
+            var out  = {tag: null, majorminor: null}
+            let tag = semver.valid(context.payload.workflow_run.head_branch)
+            if (tag == null) {
+              return out
+            } else {
+              out.tag = 'v'+tag
+            }
+            if (semver.prerelease(tag) == null) {
+              out.majorminor = 'v'+semver.major(tag)+'.'+semver.minor(tag)
+            }
+            return out
```

Comment:
It's necessary here because we won't get full ref here, only the bare branch/tag name. And docker metadata-action doesn't let us apply pattern on a custom semver value, only on git tag it discovers. This tries to replicate the exact goreleaser name template that we use for tagging docker images. The only input this requires is a branch/tag name which it gets from the github action context. 
But I get your concern, the possibility of only being able to test in production is always there. It's also exacerbated by the fact that `workflow_run` triggers can only be run once it's there on the default branch.
Let me see if there are other approaches that we can try.

Reply:
It becomes necessary due to the choice of trigger. By triggering on tag push, we don't need to write this. However, better to agree on the jira before doing anything here.